### ISSUE TITLE
Add commit for repositories that is not managed by built-in Git

### ISSUE THREAD

**Issue by oklas** [eamodio/vscode-gitlens#1135]
Hello @eamodio, thanks for great work

This is not a duplicate of #563, #375 and so on git commit requests. Where clean answer was given:

>GitLens is not a replacement for the built-in Git extension, nor is it trying to be. GitLens and the built-in Git are complementary and for some things GitLens even relies on the built-in Git to provide some features.

The `built-in git` supports only one **main repo** of the workspace (window) and does not allows to switch and commit to another repo in the workspace (window). Let me know how - if I missed. Also I read it [here](https://stackoverflow.com/a/43013150/2969544).

So to match your declaration to not to be replacement of `built-in git` let proposed feature will not allow to make a commit for **main repo** of workspace (window) - for that `built-in git` is good enough - (proposed feature will be not a replacement).

Let proposed feature feature allows to make a commit for all **others repos** (all except **main repo**) of workspace (window) - exactly that what `built-in git` does not provide. This proposed feature will not be a replacement due to it will works only with that repos which not available to work with in workspace (window) using `built-in git`. Implementing this feature will not add any possibility that duplicates possibility that provided by `built-in git` for same workspace (window). 

Opening separated workspace with another **main repo** in separated window also will match mentioned above requirement to be not a replacement because behaviour will be same relative another **main repo**.

Considering to use another workspace (window) to work with another repo become workspace feature out of sense because workspace (window) is whole things with `built-in git` working as whole thing dependent feature set and requires to use separated windows for same project and same files.

Dividing a project into different repositories (for any reason, including licensing and storage location) does not stop making the project a single whole thing, since the result of building a project is at least one whole application.

Thanks

**Comment by eamodio** [comment]
@oklas So is the main issue, that the built-in git support only supports scanning 1 level down from the root?

**Comment by oklas** [comment]
Seems there is a bug in built-in git. I have repo at root and several repos in 1 level folders of root. Built-in git shows only one repo at root. The command `open repository` does nothing.

After some experiments with ui I found a way right now how to add but only two repo. For that I need to delete `top level repo` and after that add any `1 level repo` - after that two repo appears `top repo` and `selected 1 level repo`. 

There is no difference how to add from context menu or from command palette - it works only once when repo list is empty.

Context menu from repo only allows to remove repo - not an add. Command dropdown `open repo does nothing`.

**Comment by oklas** [comment]
As of now I found some more - this is possible repo specific. Some repos may be added in any case. But some repos may lead to mentioned above behaviour.

**Comment by oklas** [comment]
Also if I remove `top level repo` when `1 level repo` present - then top level repo disappears and appears again. So there is not possible to remove `top level` repo.

**Comment by oklas** [comment]
I found the criteria why some repos may not be added. There is no internal specific in repo - the problem is just a place at `1 level`. When I move the repo outside from root (for example to user home dir) that repo become good and loaded with no problems.

**Comment by oklas** [comment]
I found simple **workaround** for this **bug** in `build-in git`. To add git a repo into workspace it requires to rename `.git` folder at root folder of workspace to any temporary name. Then add any amount of git repos (from `1 level folders` or any) into workspace. And then rename temporary git folder in root folder back to `.git`. I hope this will help someone.

**Comment by eamodio** [comment]
_Posted by @oklas_ 
> The ticket #1135 is closed for conversation but It need to have reference to currently open **microsoft/vscode** issue [#37947](https://github.com/microsoft/vscode/issues/37947).
